src/components/BSelect/BSelect.vue

```diff
@@ -0,0 +1,230 @@
+          @click="selectOption(option)"
+          role="option"
+          :ref="'option' + option.id"
+          @keyup="keyUpHandler(option)"
+          tabindex='-1'
+        > {{ option.text }}
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script>
+
+const options = []
+
+export default {
+  name: 'b-select',
+  props: {
+
+    id: {
+      type: String,
+      required: true
+    },
+    selectOptions: {
+      type: Array
+    },
+    isRequired: {
+      type: Boolean,
+      required: true
+    },
+    selectLabel: {
+      type: String,
+      required: true
+    },
+    initialValue: {
+      type: Object,
+      required: false
+    },
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Choose Option'
+    }
+  },
+
+  methods: {
+    openList () {
+      this.opened = !this.opened
+      if (this.opened) {
+        this.$nextTick( function () {
+          this.setFocus()
+        })
+      } else {
+        document.querySelector(`#dropdown${this.id}`).focus()
+      }
+    },
+    selectOption (option) {
+      this.selectedOption = option
+      this.opened = !this.opened
+      document.querySelector(`#dropdown${this.id}`).focus()
+    },
+    setFocus () {
+      if (this.selectedOption) {
+        const line = document.querySelector(`#${this.id} #option${this.selectedOption.id}`)
+        line.focus()
+      } else {
+        const line = document.querySelector(`#${this.id} #option${this.options[0].id}`)
+        line.focus()  
+      }
+    },
+    keyUpHandler (option) {
+      const target = event.target;
+      event.stopPropagation();
+      if(event.keyCode === 40) {
+        event.preventDefault();
+        if(target.nextElementSibling) {
+          const next = target.nextElementSibling;
+          next.setAttribute("tabindex", "0");
+          target.setAttribute("tabindex", "-1");
+          next.focus();
+        }
+      } else if(event.keyCode === 38) {
+        event.preventDefault();
+        if(target.previousElementSibling) {
+          const next = target.previousElementSibling;
+          const first = next.previousElementSibling;
+          if (first) {
+            next.setAttribute("tabindex", "0");
+            target.setAttribute("tabindex", "-1");
+            next.focus();
+          }
+        }
+      } else if ( event.keyCode === 32 && !this.opened) {
+        event.preventDefault();
+        this.openList()
+      } else if (event.keyCode === 32 || event.keyCode === 13) {
+        if (this.opened) {
+          event.preventDefault();
+          this.selectOption(option)
+        }
+      }
+    }
+  },
+
+  data () {
+    return {
+      options: this.selectOptions,
+      opened: false,
+      selectedOption: this.initialValue
+    }
+  },
+
+  computed: {
+    ulClass: function () {
+      return {
+        'ul-opened': this.opened
+      }
+    },
+    liClass: function () {
+      return {
+        'li-opened': this.opened,
```

Comment:
This seems redundant  since the li elements are only rendered when this.opened is true